### PR TITLE
[Concept Entry] Swift Ranges

### LINKED ISSUE #3487: [Concept Entry] Swift Ranges
### Reason/inspiration (optional)

I noticed there wasn't an entry about Ranges in Swift and thought it would be a good addition to docs.

The entry could go here: `content/swift/concepts/ranges/ranges.md`

@codecademy-docs Please could I work on this?

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [ ] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by gracekishino**
### Description

Add an entry for Swift Ranges.

Added new folder `ranges` inside `content/swift/concepts/`.
Added new entry file `content/swift/concepts/ranges/ranges.md`.

Closes #3487 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Few comments from my side for you @gracekishino to address. Then we will move to second review. :)

File: content/swift/concepts/ranges/ranges.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+`Range` instances are declared using the `..<` operator. They include values from a lower bound up to, but **excluding** an upper bound.
+
+`ClosedRange` instances are declared using the `...` operator. They contain values from a lower bound up to and **including** an upper bound.
+
+## Creating Ranges
+
+The syntax below shows how to create a `Range` with `lower` as the lower bound value and `upper` as the upper bound value:
+
+```pseudo
+let newRange = lower..<upper
```
[the remark this answers is not in the document]
Comment (gracekishino) [reply content/swift/concepts/ranges/ranges.md:26]: I changed the text to include this.

File: content/swift/concepts/ranges/ranges.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+CatalogContent:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+**Ranges** are used to create an interval of values. There are two kinds of range [structures](https://www.codecademy.com/resources/docs/swift/structures), `Range` and `ClosedRange`.
+
+`Range` instances are declared using the `..<` operator. They include values from a lower bound up to, but **excluding** an upper bound.
+
+`ClosedRange` instances are declared using the `...` operator. They contain values from a lower bound up to and **including** an upper bound.
```
[the remark this answers is not in the document]
Comment (gracekishino) [reply content/swift/concepts/ranges/ranges.md]: I added in more detail for half-open and closed ranges in a few places.

**Comment by gracekishino** [comment]
@dakshdeepHERE thanks for the review! I committed your suggestions and also added in more information about "half-open range" and "closed ranges" in a few more places.

File: content/swift/concepts/ranges/ranges.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  - 'Mobile Development'
+Tags:
+  - 'Range'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+**Ranges** are used to create an interval of values. There are two kinds of range [structures](https://www.codecademy.com/resources/docs/swift/structures), `Range` for creating half-open ranges and `ClosedRange` for creating closed ranges.
```
Comment (dakshdeepHERE) [inline content/swift/concepts/ranges/ranges.md:15]:
```suggestion
**Ranges** are used to create an interval of values. There are two kinds of range [structures](https://www.codecademy.com/resources/docs/swift/structures), `Range` for creating `half-open` ranges and `ClosedRange` for creating closed ranges.
```

File: content/swift/concepts/ranges/ranges.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+Some useful applications of ranges are:
+
+- For iteration with for-in [loops](https://www.codecademy.com/resources/docs/swift/loops).
+- In [switch](https://www.codecademy.com/resources/docs/swift/switch) statements to find out if a value lies within an interval.
+- Selecting a section of an [array](https://www.codecademy.com/resources/docs/swift/arrays).
+
+### For-in Loop Example
+
+The example below uses **for-in loops** with a half-open range and then a closed range to print some emoji trees:
```
Comment (dakshdeepHERE) [inline content/swift/concepts/ranges/ranges.md:49]:
```suggestion
The example below uses `for-in loops` with a `half-open` range and then a `closed range` to print some emoji trees:
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Thanks for the commits @gracekishino, I have two more comments for you to work on.

**Comment by gracekishino** [comment]
@dakshdeepHERE thanks for the new comments! Though I think "half-open range" and "closed range" are clearer without the formatting, is there a reason they need to have it? Also if we add the formatting to "for-in loops" then does it also need to be added to "switch", "array", etc.?

**Comment by dakshdeepHERE** [comment]
@gracekishino I formated the `half-open` because `ClosedRange` was formatted in the desc para. It was suggested just to be consistent. Also yes we use the backticks when we are referring to elements as they appear in code (properties, methods, functions, commands, expressions, etc.).

**Comment by gracekishino** [comment]
@dakshdeepHERE thanks for the clarification! 
In Swift the structure for half-open ranges is called `Range` and the structure for closed ranges is called `ClosedRange` so that's why I formatted them with backticks and didn't format the descriptive words before them. I pushed a new commit with backticks around the `for-in` loop command. 
Do you think it's OK like this?

**Comment by dakshdeepHERE** [comment]
Oh, I see @gracekishino, that makes sense. Let me go through the entry again. I think this is ready for a second review. Again an amazing entry 🚀

**Review by dakshdeepHERE — APPROVED** [review]
This is ready for second review @gracekishino. I added a tag from my side.🚀

**Review by SSwiniarski — APPROVED** [review]
@gracekishino This looks good to me.

**Comment by gracekishino** [comment]
@dakshdeepHERE and @SSwiniarski thanks so much for reviewing! 🙏

**Comment by dakshdeepHERE** [comment]
No worries @gracekishino 🫡 Amazing entry 🚀

**Pull request merged**